metsim/disaggregate.py

```diff
@@ -6,90 +6,103 @@
+from metsim.configuration import PARAMS as params
+from metsim.configuration import CONSTS as consts
 
-tiny_rad_fract = np.zeros(366) #This is updated during the mtclim run
 
-def disaggregate(df_daily):
+def disaggregate(df_daily, solar_geom):
     """
     TODO
     """
-    dates_hourly = pd.date_range(metsim.start, metsim.stop, freq='H') 
+    end = metsim.stop + pd.Timedelta('1 days')
+    dates_hourly = pd.date_range(metsim.start, end, freq='H')
     df_hourly = pd.DataFrame(index=dates_hourly)
-    _disagg_shortwave(df_daily, df_hourly)
-    _disagg_temp(     df_daily, df_hourly)
-    _disagg_precip(   df_daily, df_hourly)
-    _disagg_thermal(  df_daily, df_hourly)
-    _disagg_wind(     df_daily, df_hourly)
+    df_hourly['shortwave'] = shortwave(df_daily['swrad'],
+                                       df_daily['dayl'],
+                                       df_daily['day_of_year'],
+                                       solar_geom['tiny_rad_fract'])
+    df_hourly['temp'] = temp(df_daily, df_hourly)
+    df_hourly['precip'] = precip(df_daily['precip'])
+    df_hourly['longwave'] = longwave(df_daily['lwrad'])
+    df_hourly['wind'] = wind(df_daily['wind'])
     return df_hourly
 
-
-def _disagg_temp(df_daily, df_hourly):
+def temp(df_daily, df_hourly):
     """
     TODO
     """
     # Calculate times of min/max temps
-    set_min_max_hour(df_daily, df_hourly)
-    # Fit hermite polynomial and sample daily 
-   
-
-
-def _disagg_precip(df_daily, df_hourly):
+    hours = set_min_max_hour(df_hourly['shortwave'],
+                             len(df_daily['day_of_year']))
+    df_daily['t_Tmin'] = hours['t_Tmin']
+    df_daily['t_Tmax'] = hours['t_Tmax']
+
+    # Fit hermite polynomial and sample daily
+    # TODO: Implement this
+    # FIXME: This relies on shortwave being implemented correctly
+    # x = T_i * day_number
+    # y = [Tmin, Tmax, Tmin, Tmax, ... ]
+    # interp = scipy.interpolate.PchipInterpolator(x, y, extrapolate=True)
+    # temps = interp(range(len(df_hourly.index))
+    # return temps
+
+def precip(precip):
     """
-    TODO
+    Splits the daily precipitation evenly throughout the day
     """
-    pass
+    return precip.resample('H', how='sum').fillna(method='ffill')/24.
 
 
-def _disagg_thermal(df_daily, df_hourly):
+def longwave(longwave):
     """
-    TODO
+    Splits the daily longwave evenly throughout the day
     """
-    pass
+    return longwave.resample('H', how='sum').fillna(method='ffill')/24.
 
 
-def _disagg_wind(df_daily, df_hourly):
+def wind(wind):
     """
-    TODO
-    """   
-    pass
+    Wind is assumed constant throughout the day
+    """
+    return wind.resample('H', how='sum').fillna(method='ffill')
 
 
-def _disagg_shortwave(df_daily, df_hourly):
+def shortwave(sw_rad, daylength, day_of_year, tiny_rad_fract):
     """
     TODO
     """
     tiny_step_per_hour = int(3600 / consts['SRADDT'])
-    tmp_rad = df_daily['s_swrad']
+    tmp_rad = sw_rad * daylength / 3600.
```

Comment:
Is `sw_rad` a cumulative flux or a mean flux? If the latter than this does not work.